controllers/admin/AdminImportController.php

```diff
@@ -3762,44 +3797,66 @@ public function storeContactImportOne($info, $shop_is_feature_active, $regenerat
                 $state->id_zone = 0; // Default zone for state to create
                 $state->iso_code = Tools::strtoupper(Tools::substr($store->state, 0, 2)); // Default iso for state to create
                 $state->tax_behavior = 0;
-                if (($field_error = $state->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-                    ($lang_field_error = $state->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
+                if (($fieldsValidationResult = $state->validateFields(UNFRIENDLY_ERROR, true)) === true &&
+                    ($langFieldsValidationResult = $state->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
                     !$validateOnly && // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
                     $state->add()) {
                     $store->id_state = (int)$state->id;
                 } else {
                     if (!$validateOnly) {
                         $this->errors[] = sprintf($this->trans('%s cannot be saved', array(), 'Admin.Advparameters.Notification'), $state->name);
                     }
-                    if ($field_error !== true || isset($lang_field_error) && $lang_field_error !== true) {
-                        $this->errors[] = ($field_error !== true ? $field_error : '').(isset($lang_field_error) && $lang_field_error !== true ? $lang_field_error : '').
+                    if ($fieldsValidationResult !== true || isset($langFieldsValidationResult) && $langFieldsValidationResult !== true) {
+                        $this->errors[] = ($fieldsValidationResult !== true ? $fieldsValidationResult : '').(isset($langFieldsValidationResult) && $langFieldsValidationResult !== true ? $langFieldsValidationResult : '').
                             Db::getInstance()->getMsgError();
                     }
                 }
             }
         }
 
-        $res = false;
-        if (($field_error = $store->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-            ($lang_field_error = $store->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true) {
-            if ($store->id && $store->storeExists($store->id)) {
-                $res = $validateOnly ? $validateOnly : $store->update();
+        $fieldsValidationResult     = $store->validateFields(UNFRIENDLY_ERROR, true);
+        $langFieldsValidationResult = $store->validateFieldsLang(UNFRIENDLY_ERROR, true);
+
+        // If errors, log them and stop execution
+        if (true !== $fieldsValidationResult || true !== $langFieldsValidationResult) {
+            $errorGenericMessage = $this->trans('Store is invalid', [], 'Admin.Advparameters.Notification');
```

Comment:
It does not grind our gears like for Peter Griffin, but the rules are the same for everybody. :wink: